Firebase/android/app/src/main/java/com/novoda/bonfire/FirebaseConfig.java

```diff
@@ -0,0 +1,58 @@
+package com.novoda.bonfire;
+
+import android.support.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.remoteconfig.FirebaseRemoteConfig;
+import com.google.firebase.remoteconfig.FirebaseRemoteConfigSettings;
+import com.novoda.bonfire.analytics.ErrorLogger;
+
+public class FirebaseConfig implements Config {
+
+    private static final String ORDER_CHANNELS_BY_NAME = "orderChannelsByName";
+    private static final int CACHE_EXPIRATION_IN_SECONDS = 3600;
+
+    private final FirebaseRemoteConfig firebaseRemoteConfig;
+
+    public static FirebaseConfig newInstance() {
+        final FirebaseRemoteConfig firebaseRemoteConfig = FirebaseRemoteConfig.getInstance();
+        FirebaseRemoteConfigSettings configSettings = new FirebaseRemoteConfigSettings.Builder()
+                .setDeveloperModeEnabled(BuildConfig.DEBUG)
+                .build();
+        firebaseRemoteConfig.setConfigSettings(configSettings);
+        firebaseRemoteConfig.setDefaults(R.xml.remote_config_defaults);
+        return new FirebaseConfig(firebaseRemoteConfig);
+    }
+
+    public FirebaseConfig init(final ErrorLogger errorLogger) {
+        firebaseRemoteConfig.fetch(CACHE_EXPIRATION_IN_SECONDS)
```

Comment:
Cache expiry configuration, it will check for new values if the cached value is older than this number.